go-controller/pkg/node/gateway_udn.go

```diff
@@ -0,0 +1,154 @@
+package node
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+	utilnet "k8s.io/utils/net"
+)
+
+// UserDefinedNetworkGateway contains information
+// required to program a UDN at each node's
+// gateway.
+// NOTE: Currently invoked only for primary networks.
+type UserDefinedNetworkGateway struct {
+	// network information
+	util.NetInfo
+	// stores the networkID of this network
+	networkID int
+	// node that its programming things on
+	node *v1.Node
+}
+
+func NewUserDefinedNetworkGateway(netInfo util.NetInfo, networkID int, node *v1.Node) *UserDefinedNetworkGateway {
+	return &UserDefinedNetworkGateway{
+		NetInfo:   netInfo,
+		networkID: networkID,
+		node:      node,
+	}
+}
+
+// AddNetwork will be responsible to create all plumbings
+// required by this UDN on the gateway side
+func (udng *UserDefinedNetworkGateway) AddNetwork() error {
+	return udng.addUDNManagementPort()
+}
+
+// DelNetwork will be responsible to remove all plumbings
+// used by this UDN on the gateway side
+func (udng *UserDefinedNetworkGateway) DelNetwork() error {
+	return udng.deleteUDNManagementPort()
+}
+
+// addUDNManagementPort does the following:
+// STEP1: creates the OVS interface on br-int for the UDN's management port interface
+// STEP2: It saves the MAC address generated on the 1st go as an option on the OVS interface
+// so that it persists on reboots
+// STEP3: creates the management port netdevice on the host
+// STEP4: adds the management port IP .2 to the netdevice
+func (udng *UserDefinedNetworkGateway) addUDNManagementPort() error {
+	var err error
+	interfaceName := util.GetNetworkScopedK8sMgmtHostIntfName(uint(udng.networkID))
+	var networkLocalSubnets []*net.IPNet
+	if udng.TopologyType() == types.Layer3Topology {
+		networkLocalSubnets, err = util.ParseNodeHostSubnetAnnotation(udng.node, udng.GetNetworkName())
+		if err != nil {
+			return fmt.Errorf("waiting for node %s to start, no annotation found on node for network %s: %v",
+				udng.node.Name, udng.GetNetworkName(), err)
+		}
+	} else if udng.TopologyType() == types.Layer2Topology {
+		globalFlatL2Networks := udng.Subnets()
+		for _, globalFlatL2Network := range globalFlatL2Networks {
+			networkLocalSubnets = append(networkLocalSubnets, globalFlatL2Network.CIDR)
+		}
+	}
+	networkMTU := udng.NetInfo.MTU()
+	if networkMTU == 0 {
+		networkMTU = config.Default.MTU
+	}
```

Comment:
When parsing netconf it's already defaulting to proper value so we jsut need to do 
```golang
networkMTU := udng.NetInfo.MTU()
```

This is the parsing mechanism ovn-k has 
```golang
func parseNetConfSingle(bytes []byte) (*ovncnitypes.NetConf, error) {           
    netconf := &ovncnitypes.NetConf{MTU: Default.MTU}                           
    err := json.Unmarshal(bytes, &netconf)                                      
    if err != nil {                                                             
        return nil, err                                                         
    }                                                                           
                                                                                
    // skip non-OVN NAD                                                         
    if netconf.Type != "ovn-k8s-cni-overlay" {                                  
        return nil, ErrorAttachDefNotOvnManaged                                 
    }                                                                           
                                                                                
    err = validateNetConfNameFields(netconf)                                    
    if err != nil {                                                             
        return nil, err                                                         
    }                                                                           
                                                                                
    return netconf, nil                                                         
}             
```

Reply:
oh neat! so its already picking up config.Default.MTU if not set by user via NAD?

Reply:
It's also unit tested, at the util multi_homing we have unit tests like 
```golang
      {                                                                       
            desc: "valid attachment definition for a localnet topology with a VLAN",
            inputNetAttachDefConfigSpec: `                                      
    {                                                                           
            "name": "tenantred",                                                
            "type": "ovn-k8s-cni-overlay",                                      
            "topology": "localnet",                                             
            "vlanID": 10,                                                       
            "netAttachDefName": "ns1/nad1"                                      
    }                                                                           
`,                                                                              
            expectedNetConf: &ovncnitypes.NetConf{                              
                Topology: "localnet",                                           
                NADName:  "ns1/nad1",                                           
                MTU:      1400,                                                 
                VLANID:   10,                                                   
                NetConf:  cnitypes.NetConf{Name: "tenantred", Type: "ovn-k8s-cni-overlay"},
            },                                                                  
        },                                                                      
```

At golang when you don't specify an integer it's value is `0`.

Reply:
woohoo ok changing this on my side